### PR TITLE
[shopsys] automatically generated entity extension map

### PR DESCRIPTION
**Pull request by pk16011990**
| Q             | A
| ------------- | ---
|Description, reason for the PR| It saves time and prevents bugs. It was annoying and unnecessary.
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by PetrHeinz** [comment]
Pretty cool idea @pk16011990! This will help a lot. Somebody should go through the docs and explain the rules (hardcoded namespace prefixes `App\` and `Shopsys\ShopsysFramework\`) and update (simplify, in most cases) the articles mentioning `shopsys.entity_extension.map`.

File: docs/extensibility/entity-extension.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ This article describes a quick way to extend your entity and the internals of en
         * `User`
     * keep entity and table annotations
     * you can add new properties and use annotations to configure ORM
-* Add information about the entity extension into the container configuration
+* Entities from namespace `App\` extending entities from namespace `Shopsys\` are automatically extended, if you want to extend from or to different namespace you need to add information about the entity extension into the container configuration
     * add it to the configuration parameter `shopsys.entity_extension.map` placed in `config/parameters_common.yml` file
     * use the parent entity name as a key and the extended entity name as a value
     * eg. `Shopsys\FrameworkBundle\Model\Product\Product: App\Model\Product\Product`
```
Comment (grossmannmartin) [inline docs/extensibility/entity-extension.md:24]: What about to change the example to something that is not covered with compiler pass to be exact?

File: packages/framework/src/DependencyInjection/Compiler/RegisterExtendedEntitiesCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class RegisterExtendedEntitiesCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $container->initialized('doctrine.orm.default_metadata_driver');
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply packages/framework/src/DependencyInjection/Compiler/RegisterExtendedEntitiesCompilerPass.php:18]: You are right, removed

File: packages/framework/src/DependencyInjection/Compiler/RegisterExtendedEntitiesCompilerPass.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                $parentClass = get_parent_class($class);
+
+                if ($parentClass !== false && strpos($parentClass, 'Shopsys\\') === 0) {
+                    $annotationReader->loadMetadataForClass($parentClass, new ClassMetadata($parentClass));
+                    $entityExtensionMap[$parentClass] = $class;
+                }
+            }
+        }
+
+        $container->setParameter('shopsys.entity_extension.map', $entityExtensionMap);
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply packages/framework/src/DependencyInjection/Compiler/RegisterExtendedEntitiesCompilerPass.php:36]: I agree, fixed

File: docs/extensibility/entity-extension.md (hunk 1 above)
Comment (TomasLudvik) [reply docs/extensibility/entity-extension.md:24]: Updated

**Pull request merged**